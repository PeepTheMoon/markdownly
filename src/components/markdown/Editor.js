import React from 'react';
import styles from './Editor.css';
import { useDispatch, useSelector } from '../../hooks/Context';
import { editMarkdown } from '../../actions/reducerActions';
import { getMarkdown } from '../../selectors/selectors';

const Editor = () => {
  const dispatch = useDispatch();
  const markdown = useSelector(getMarkdown);
  
  const handleChange = ({ target }) => {
    dispatch(editMarkdown(target.value));
  };

  return (
    <textarea className={styles.Editor} value={markdown} onChange={handleChange} />
  );
};

export default Editor;
