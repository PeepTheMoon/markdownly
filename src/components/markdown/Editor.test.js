import React from 'react';
import { shallow } from 'enzyme';
import Editor from './Editor';

jest.mock('../../hooks/Context');

describe('Editor component', () => {
  it('renders editor', () => {
    const wrapper = shallow(<Editor />);
    expect(wrapper).toMatchSnapshot();
  });
});
