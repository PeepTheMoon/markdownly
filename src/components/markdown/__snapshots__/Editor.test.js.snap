// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor component renders editor 1`] = `
<textarea
  className="Editor"
  onChange={[Function]}
/>
`;
